srcbpatch/actionscollection.cpp

```diff
@@ -0,0 +1,488 @@
+        if (!exists)
+        {
+            pTarget = nullptr;
+            break; // not a string received but object
+        }
+
+        if (0 == i)
+        { // save the pointer on first element - here we will accumulate the data
+            pTarget = const_cast<char*>(value.data());
+        }
+
+        unsigned char c = 0;
+        std::from_chars_result rez =
+            std::from_chars(value.data(), value.data() + value.length(),
+                c, base);
+        if (rez.ec != std::errc())
+        { // we got something which is not a number
+            pTarget = nullptr;
+            break;
+        }
+        // add a value to the result string view
+        pTarget[rezLength++] = *reinterpret_cast<char*>(&c);
+    }
+
+    if (nullptr == pTarget)
+    {
+        ReportError(errorMsg); // throws
+        return emptySV;
+    }
+    else
+    {
+        return std::string_view(pTarget, rezLength);
+    }
+}
+
+
+std::vector<std::string_view> ActionsCollection::ParseArray(TJSONObject* const pJson, const char* const errorMsg)
+{
+    const size_t sz = pJson->ChildCount();
+    if (sz == 0) // empty array
+    {
+        ReportError(errorMsg);
+    }
+
+    std::vector<std::string_view> ret;
+    for (size_t i = 0; i < sz; ++i)
+    {
+        auto [exists, value] = pJson->JsonValue(i);
+        if (!exists)
+        {
+            ReportError(errorMsg);// not a string received but object
+        }
+        ret.emplace_back(std::move(value));
+    }
+
+    return ret;
+}
+
+
+void ActionsCollection::ProcessComposites()
+{
+    for (const auto& [name, vectorData]: composites_)
+    {// for every composite lexeme
+
+        std::unique_ptr<AbstractBinaryLexeme> compositeLexeme;
+        size_t totalLength = 0;
+        for (const auto& subname : vectorData)
+        {
+            // get existent lexeme
+            auto& existantLexeme = dictionary_.Lexeme(subname);
+            if (existantLexeme == nullptr)
+            {
+                ReportMissedNameError(subname);
+            }
+            const size_t len = existantLexeme->access().size();
+
+            if (compositeLexeme == nullptr) // just created
+            {
+                compositeLexeme = 
+                    AbstractBinaryLexeme::LexemeFromLexeme(existantLexeme);
+                totalLength = len;
+                continue;
+            }
+
+            // fill data for composite lexeme
+            compositeLexeme->Replace(totalLength, 0, existantLexeme);
+            totalLength += len;
+        }
+
+        // now move newly created lexeme into Dictionary
+        if (const bool added = 
+            dictionary_.AddBinaryLexeme(name, std::move(compositeLexeme));
+            !added)
+        {// overwritten
+            ReportDuplicateNameError(name);
+        }
+    }
+
+    // everything has been processd.
+    // free unnecessary collection
+    composites_.clear();
+}
+
+
+
+
+void ActionsCollection::CreateChainOfReplacers()
```

Comment:
I suggest you to decouple data validation and data processing. I don't think if you go through the json file one more time it makes a huge impact on performance. But I agree that it's debatable